scio-bigtable/src/main/scala/com/spotify/scio/bigtable/TableAdmin.scala

```diff
@@ -0,0 +1,113 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.spotify.scio.bigtable
+
+import com.google.bigtable.admin.v2._
+import com.google.bigtable.admin.v2.ModifyColumnFamiliesRequest.Modification
+import com.google.cloud.bigtable.config.BigtableOptions
+import com.google.cloud.bigtable.grpc.BigtableTableAdminGrpcClient
+import org.slf4j.{Logger, LoggerFactory}
+
+import scala.collection.JavaConverters._
+
+/**
+ * Bigtable Table Admin API helper commands.
+ */
+object TableAdmin {
+  val log: Logger = LoggerFactory.getLogger(TableAdmin.getClass)
+
+  /**
+   * Ensure that tables and column families exist.
+   * Checks for existence of tables or creates them if they don't exist.
+   * Then calls `ensureColumnFamilies` to ensure that the column families exist as well.
+   *
+   * @param tablesAndColumnFamilies A map of tables and column families.  Keys are table names,
+   *                                Values are a list of column family names.
+   */
+  def ensureTables(bigtableOptions: BigtableOptions,
+                   tablesAndColumnFamilies: Map[String, List[String]]): Unit = {
+
+    val channel = ChannelPoolCreator.createPool(bigtableOptions.getTableAdminHost)
+    val client = new BigtableTableAdminGrpcClient(channel)
+    val project = bigtableOptions.getProjectId
+    val instance = bigtableOptions.getInstanceId
+    val instancePath = s"projects/$project/instances/$instance"
+
+    log.info("Ensuring tables and column families exist in instance {}", instance)
+
+    try {
+      val existingTables = client.listTables(ListTablesRequest.newBuilder()
+        .setParent(instancePath)
+        .build())
+        .getTablesList.asScala
+        .map(t => t.getName).toSet
+
+      for ((table, columnFamilies) <- tablesAndColumnFamilies) {
+        val tablePath = s"$instancePath/tables/$table"
+
+        if (!existingTables.contains(tablePath)) {
+          log.info("Creating table {}", table)
+          client.createTable(CreateTableRequest.newBuilder()
+            .setParent(instancePath)
+            .setTableId(table)
+            .build())
+        } else {
+          log.info("Table {} exists", table)
+        }
+
+        ensureColumnFamilies(client, tablePath, columnFamilies)
+      }
+    } finally {
+      channel.shutdownNow()
+    }
+  }
+
+  /**
+   * Ensure that column families exist.
+   * Checks for existence of column families and creates them if they don't exist.
+   *
+   * @param tablePath A full table path that the bigtable API expects, in the form of
+   *                  `projects/&lt;projectid&gt;/instances/&lt;instance&gt;/tables/&lt;table&gt;`
```

Comment:
Ugh, so $name gets interpreted as a scaladoc macro http://docs.scala-lang.org/overviews/scaladoc/for-library-authors.html#macros and there doesn't appear to be an escape sequence.  I'm just going to remove both all symbols and since this fn will be private, it won't end up in the generated docs anyway.